Phylanx build fails

Phylanx build fails on CircleCI and it seems to complain about HighFive.
This is the error:
```console
[168/303] Building CXX object src/plugins/fileio/CMakeFiles/fileio_primitive.dir/file_read_hdf5.cpp.o
FAILED: src/plugins/fileio/CMakeFiles/fileio_primitive.dir/file_read_hdf5.cpp.o 
/usr/bin/clang++  -DBLAZE_USE_CPP_THREADS -DBOOST_ALL_NO_LIB -DDEBUG -DHPX_DEBUG -DHPX_ENABLE_ASSERT_HANDLER -DHPX_LIBRARY_EXPORTS -DHPX_PLUGIN_NAME=phylanx_fileio -DHPX_PLUGIN_STRING=\"phylanx_fileio\" -DMPI_NO_CPPBIND -DPHYLANX_LIBRARY_EXPORTS -D_DEBUG -Dfileio_primitive_EXPORTS -I/phylanx/src -I/usr/local/include/BlazeIterative -isystem /usr/include/hdf5/serial -I. -isystem /usr/local/include -fdiagnostics-show-option -Werror=vla -Werror=return-type -Wno-cast-align -g -fPIC   -std=c++17 -std=gnu++1z -MD -MT src/plugins/fileio/CMakeFiles/fileio_primitive.dir/file_read_hdf5.cpp.o -MF src/plugins/fileio/CMakeFiles/fileio_primitive.dir/file_read_hdf5.cpp.o.d -o src/plugins/fileio/CMakeFiles/fileio_primitive.dir/file_read_hdf5.cpp.o -c /phylanx/src/src/plugins/fileio/file_read_hdf5.cpp
In file included from /phylanx/src/src/plugins/fileio/file_read_hdf5.cpp:17:
In file included from /phylanx/src/phylanx/util/detail/blaze-highfive.hpp:12:
In file included from /usr/local/include/highfive/H5File.hpp:76:
In file included from /usr/local/include/highfive/H5Group.hpp:32:
In file included from /usr/local/include/highfive/bits/H5Node_traits_misc.hpp:23:
In file included from /usr/local/include/highfive/H5DataSet.hpp:102:
In file included from /usr/local/include/highfive/bits/../H5Selection.hpp:64:
/usr/local/include/highfive/bits/H5Slice_traits_misc.hpp:170:32: error: no matching constructor for initialization of 'details::data_converter<DynamicVector<double, false> >'
    details::data_converter<T> converter(mem_space);
                               ^         ~~~~~~~~~
/phylanx/src/src/plugins/fileio/file_read_hdf5.cpp:105:25: note: in instantiation of function template specialization 'HighFive::SliceTraits<HighFive::DataSet>::read<blaze::DynamicVector<double, false> >' requested here
                dataSet.read(vector);
                        ^
/usr/local/include/highfive/bits/H5Converter_misc.hpp:122:8: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const HighFive::DataSpace' to 'const HighFive::details::data_converter<blaze::DynamicVector<double, false>, void>' for 1st argument
struct data_converter {
       ^
/usr/local/include/highfive/bits/H5Converter_misc.hpp:122:8: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'const HighFive::DataSpace' to 'HighFive::details::data_converter<blaze::DynamicVector<double, false>, void>' for 1st argument
/phylanx/src/phylanx/util/detail/blaze-highfive.hpp:82:12: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
    inline data_converter(Vector& vector, DataSpace& space, std::size_t dim = 0)
           ^
In file included from /phylanx/src/src/plugins/fileio/file_read_hdf5.cpp:17:
In file included from /phylanx/src/phylanx/util/detail/blaze-highfive.hpp:12:
In file included from /usr/local/include/highfive/H5File.hpp:76:
In file included from /usr/local/include/highfive/H5Group.hpp:32:
In file included from /usr/local/include/highfive/bits/H5Node_traits_misc.hpp:23:
In file included from /usr/local/include/highfive/H5DataSet.hpp:102:
In file included from /usr/local/include/highfive/bits/../H5Selection.hpp:64:
/usr/local/include/highfive/bits/H5Slice_traits_misc.hpp:170:32: error: no matching constructor for initialization of 'details::data_converter<DynamicMatrix<double, false> >'
    details::data_converter<T> converter(mem_space);
                               ^         ~~~~~~~~~
/phylanx/src/src/plugins/fileio/file_read_hdf5.cpp:115:25: note: in instantiation of function template specialization 'HighFive::SliceTraits<HighFive::DataSet>::read<blaze::DynamicMatrix<double, false> >' requested here
                dataSet.read(matrix);
                        ^
/usr/local/include/highfive/bits/H5Converter_misc.hpp:122:8: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const HighFive::DataSpace' to 'const HighFive::details::data_converter<blaze::DynamicMatrix<double, false>, void>' for 1st argument
struct data_converter {
       ^
/usr/local/include/highfive/bits/H5Converter_misc.hpp:122:8: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'const HighFive::DataSpace' to 'HighFive::details::data_converter<blaze::DynamicMatrix<double, false>, void>' for 1st argument
/phylanx/src/phylanx/util/detail/blaze-highfive.hpp:150:12: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
    inline data_converter(Matrix& matrix, DataSpace& space, std::size_t dim = 0)
           ^
2 errors generated.
ninja: build stopped: subcommand failed.

Exited with code exit status 1
```

--- a/phylanx/util/detail/blaze-highfive.hpp
+++ b/phylanx/util/detail/blaze-highfive.hpp
@@ -79,30 +79,28 @@ struct data_converter<blaze::DynamicVector<T, TF>, void>
 {
     typedef typename blaze::DynamicVector<T, TF> Vector;
 
-    inline data_converter(Vector& vector, DataSpace& space, std::size_t dim = 0)
+    inline data_converter(DataSpace const& space)
             : _dims(space.getDimensions())
     {
         assert(_dims.size() == 1);
-        (void) dim;
-        (void) vector;
     }
 
-    inline typename type_of_array<T>::type* transform_read(Vector& vector)
+    inline typename type_of_array<T>::type* transform_read(Vector& vector) const
     {
         if (_dims[0] != vector.size())
         {
-            std::cout << "resizing, fun!" << std::endl;
             vector.resize(_dims[0]);
         }
         return vector.data();
     }
 
-    inline typename type_of_array<T>::type* transform_write(Vector& vector)
+    inline typename type_of_array<T>::type const* transform_write(
+        Vector const& vector) const noexcept
     {
         return vector.data();
     }
 
-    inline void process_result(Vector& vector)
+    inline void process_result(Vector& vector) const
     {
         (void) vector;
     }
@@ -116,25 +114,24 @@ struct data_converter<blaze::CustomVector<T, AF, PF, TF, RT>, void>
 {
     typedef typename blaze::CustomVector<T, AF, PF, TF, RT> Vector;
 
-    inline data_converter(Vector& vector, DataSpace& space, std::size_t dim = 0)
+    inline data_converter(DataSpace const& space)
             : _dims(space.getDimensions())
     {
         assert(_dims.size() == 1);
-        (void) dim;
-        (void) vector;
     }
 
-    inline typename type_of_array<T>::type* transform_read(Vector& vector)
+    inline typename type_of_array<T>::type* transform_read(Vector& vector) const
     {
         throw std::runtime_error("can't read into blaze::CustomVector");
     }
 
-    inline typename type_of_array<T>::type* transform_write(Vector& vector)
+    inline typename type_of_array<T>::type const* transform_write(
+        Vector const& vector) const noexcept
     {
         return vector.data();
     }
 
-    inline void process_result(Vector& vector)
+    inline void process_result(Vector& vector) const
     {
         (void) vector;
     }
@@ -147,30 +144,28 @@ struct data_converter<blaze::DynamicMatrix<T, SO>, void>
 {
     typedef typename blaze::DynamicMatrix<T, SO> Matrix;
 
-    inline data_converter(Matrix& matrix, DataSpace& space, std::size_t dim = 0)
+    inline data_converter(DataSpace const& space)
       : _dims(space.getDimensions())
     {
         assert(_dims.size() == 2);
-        (void) dim;
-        (void) matrix;
     }
 
-    inline typename type_of_array<T>::type* transform_read(Matrix& matrix)
+    inline typename type_of_array<T>::type* transform_read(Matrix& matrix) const
     {
         if (_dims[0] != matrix.rows() || _dims[1] != matrix.columns())
         {
-            std::cout << "resizing, fun!" << std::endl;
             matrix.resize(_dims[0], _dims[1]);
         }
         return matrix.data();
     }
 
-    inline typename type_of_array<T>::type* transform_write(Matrix& matrix)
+    inline typename type_of_array<T>::type const* transform_write(
+        Matrix const& matrix) const noexcept
     {
         return matrix.data();
     }
 
-    inline void process_result(Matrix& matrix)
+    inline void process_result(Matrix& matrix) const
     {
         (void) matrix;
     }
@@ -184,25 +179,25 @@ struct data_converter<blaze::CustomMatrix<T, AF, PF, SO, RT>, void>
 {
     typedef typename blaze::CustomMatrix<T, AF, PF, SO, RT> Matrix;
 
-    inline data_converter(Matrix& matrix, DataSpace& space, std::size_t dim = 0)
+    inline data_converter(DataSpace const& space)
       : _dims(space.getDimensions())
     {
         assert(_dims.size() == 2);
-        (void) dim;
-        (void) matrix;
     }
 
-    inline typename type_of_array<T>::type* transform_read(Matrix& matrix)
+    inline typename type_of_array<T>::type* transform_read(
+        Matrix& matrix) const
     {
         throw std::runtime_error("can't read into blaze::CustomMatrix");
     }
 
-    inline typename type_of_array<T>::type* transform_write(Matrix& matrix)
+    inline typename type_of_array<T>::type const* transform_write(
+        Matrix const& matrix) const noexcept
     {
         return matrix.data();
     }
 
-    inline void process_result(Matrix& matrix)
+    inline void process_result(Matrix& matrix) const
     {
         (void) matrix;
     }

